src/main/java/core/basesyntax/JavaDateTimeApi.java

```diff
@@ -1,125 +1,169 @@
 package core.basesyntax;
 
+import java.time.DateTimeException;
 import java.time.LocalDate;
 import java.time.LocalDateTime;
 import java.time.LocalTime;
 import java.time.OffsetDateTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.Locale;
 import java.util.Optional;
 
 public class JavaDateTimeApi {
+    private static final String timezoneUkraine = "+02:00";
+    private static final String FORMAT_D_MMM_YYYY = "d MMM yyyy";
+    private static final String FORMAT_DD_MMMM_YYYY_HH_MM = "dd MMMM yyyy HH:mm";
+
     /**
      * Return the current date as a String depending on a query.
-     *
+     * <p>
      * The query can be passed for the whole date or for it's part:
-     *      - FULL - current date as a whole: year, month, day of month
-     *        formatted as `YYYY-MM-DD` (a default return value);
-     *      - YEAR - current year;
-     *      - MONTH - name of the current month;
-     *      - DAY - current day of month;
+     * - FULL - current date as a whole: year, month, day of month
+     * formatted as `YYYY-MM-DD` (a default return value);
+     * - YEAR - current year;
+     * - MONTH - name of the current month;
+     * - DAY - current day of month;
      * In any other case throw DateTimeException.
      **/
     public String todayDate(DateTimePart datePart) {
-        return "Today";
+        LocalDate localDate = LocalDate.now();
```

Comment:
Common mistakes
Avoid calling LocalDate.now multiple times in one method. Let’s create a variable for this purpose.

Reply:
I think it means not to write here something like that
`

        switch (datePart) {
            case DAY:
                return String.valueOf(LocalDate.now().getDayOfMonth());
            case FULL:
                return String.valueOf(LocalDate.now());
            case YEAR:
                return String.valueOf(LocalDate.now().getYear());
            case MONTH:
                return String.valueOf(LocalDate.now().getMonth());
            default:
                throw new DateTimeException("Incorrect input");`